src/main/java/producer/RecordBatch.java

```diff
@@ -0,0 +1,62 @@
+package producer;
+
+import org.tinylog.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RecordBatch {
+    private final int maxBatchSizeInBytes;
+    private int currBatchSize;
```

Comment:
Minor nit: maybe this can be `currBatchSizeInBytes`. What are your thoughts?

Reply:
I agree with this suggestion, it consistent with `maxMatchSizeInBytes` prefix